apps/src/templates/sectionProgressV2/InviteToV2ProgressModal.jsx

```diff
@@ -0,0 +1,140 @@
+import {connect} from 'react-redux';
+
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import {Heading2, BodyTwoText} from '@cdo/apps/componentLibrary/typography';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
+import {default as LinkedButton} from '@cdo/apps/templates/Button';
+import i18n from '@cdo/locale';
+
+import styles from './progress-v2-invitation.module.scss';
+
+const newProgressViewGraphic = require('@cdo/static/teacherDashboard/progressOpenBetaAnnouncementGraphic.png');
+
+function InviteToV2ProgressModal({
+  setShowProgressTableV2,
+  sectionId,
+
+  // from redux
+  dateProgressTableInvtationDelayed,
+  hasSeenProgressTableInvite,
+}) {
+  const [invitationOpen, setInvitationOpen] = React.useState(false);
+
+  React.useEffect(() => {
+    const timeSinceInvitationLastDelayed = () => {
+      const startingDate = new Date(dateProgressTableInvtationDelayed);
+      const today = new Date();
+      const differenceInMilliseconds = today.getTime() - startingDate.getTime();
+      const differenceInDays = differenceInMilliseconds / (1000 * 3600 * 24);
+      return Math.floor(differenceInDays);
+    };
+
+    const showInvitation = () => {
+      const alreadyViewedInvitation = !!hasSeenProgressTableInvite;
+      if (alreadyViewedInvitation) {
+        return false;
+      } else {
+        if (!!dateProgressTableInvtationDelayed) {
+          return timeSinceInvitationLastDelayed() > 3;
+        } else {
+          return true;
+        }
+      }
+    };
+
+    setInvitationOpen(showInvitation());
+  }, [dateProgressTableInvtationDelayed, hasSeenProgressTableInvite]);
+
+  const handleDismiss = React.useCallback(() => {
+    analyticsReporter.sendEvent(EVENTS.PROGRESS_V2_DISMISS_INVITATION, {
+      sectionId,
+    });
+    setHasSeenProgressTableInvite();
+    setInvitationOpen(false);
+  }, [sectionId]);
+
+  const handleAcceptedInvitation = React.useCallback(() => {
+    analyticsReporter.sendEvent(EVENTS.PROGRESS_V2_ACCEPT_INVITATION, {
+      sectionId,
+    });
+    setHasSeenProgressTableInvite();
+    setShowProgressTableV2(true);
+  }, [sectionId, setShowProgressTableV2]);
+
+  const handleDelayInvitation = React.useCallback(() => {
+    analyticsReporter.sendEvent(EVENTS.PROGRESS_V2_DELAY_INVITATION, {
+      sectionId,
+    });
+    setDateInvitationDelayed(new Date());
+    setInvitationOpen(false);
+  }, [sectionId]);
+
+  const setDateInvitationDelayed = date => {
+    return $.post(`/api/v1/users/date_progress_table_invitation_last_delayed`, {
+      date_progress_table_invitation_last_delayed: date,
+    });
+  };
+
+  const setHasSeenProgressTableInvite = () => {
+    return $.post(`/api/v1/users/has_seen_progress_table_v2_invitation`, {
+      has_seen_progress_table_v2_invitation: true,
+    });
+  };
+
+  if (invitationOpen) {
```

Comment:
Optional nit: I find `  if (!invitationOpen) { return null} ...` to be slightly clearer because it shows the early return before the big blob of content to show. Completely optional nit pick